File: Python/LazyInitialization.py
class Lazy():
	def __init__(self, *data):
		self._data = data;
		self._initialized = False;

	def lazyInit(self):
		self.var = " ".join(self._data);
		self._initialized = True;
	
	def callMe(self):
		if not self._initialized:
			lazyInit();
		print(self.var);

File: Java/LazyInitialization.java
public class Lazy {
	private boolean initialized;
	private String[] data;
	public String var;
	
	public Lazy(String... data) {
		this.data = data;
		this.initialized = false;
	}

	public void lazyInit() {
		this.var = String.join(" ", this.data);
		this.initialized = true;
	}
	
	public void callMe() {
		if (!this.initialized){
			lazyInit();
		}
		System.out.println(this.var);
}